File: app/src/main/java/com/hazizz/droid/Communication/Requests/RequestType/Thera/ThReturnSchedules/PojoClass.java
package com.hazizz.droid.Communication.Requests.RequestType.Thera.ThReturnSchedules;

import lombok.Data;

@Data
public class PojoClass {

    String date;
    String startOfClass;
    String endOfClass;
    int periodNumber;
    boolean cancelled;
    boolean standIn;
    String subject;
    String className;
    String teacher;
    String room;
    String topic;

    public PojoClass(String date, String startOfClass, String endOfClass, int periodNumber, boolean cancelled, boolean standIn,String subject, String className, String teacher, String room, String topic) {
        this.date = date;
        this.startOfClass = startOfClass;
        this.endOfClass = endOfClass;
        this.periodNumber = periodNumber;
        this.cancelled = cancelled;
        this.standIn = standIn;
        this.subject = subject;
        this.className = className;
        this.teacher = teacher;
        this.room = room;
        this.topic = topic;
    }
}

File: lib/communication/pojos/PojoClass.dart
import 'package:flutter/material.dart';
import 'package:mobile/custom/hazizz_logger.dart';
import 'Pojo.dart';

part 'PojoClass.gg.dart';

// removing so it wont recreate
//@JsonSerializable()
class PojoClass extends Pojo{
  String accountId;
  DateTime date;
  TimeOfDay startOfClass;
  TimeOfDay endOfClass;
  int periodNumber;
  bool cancelled;
  bool standIn;
  String subject;
  String className;
  String teacher;
  String room;
  String topic;

  PojoClass({this.accountId, this.date, this.startOfClass, this.endOfClass, this.periodNumber,
      this.cancelled, this.standIn, this.subject, this.className, this.teacher,
      this.room, this.topic});

  factory PojoClass.fromJson(Map<String, dynamic> json){
    PojoClass pojoClass = _$PojoClassFromJson(json);
   // pojoClass.className = "MatekMatekMatek";
    return pojoClass;
  }

  Map<String, dynamic> toJson() => _$PojoClassToJson(this);

}